content/articles/visualizing-repositories-using-pygal/index.md

```diff
@@ -0,0 +1,371 @@
+
+respons_dict = request.json()  # In a variable, save the API response.
+
+print("Total repos:", respons_dict['total_count'])
+
+# Examine the repository's details.
+
+repos_dicts = respons_dict['items']
+
+print("Repos found:", len(repos_dicts))
+
+print("\nSelected info about each repository:")
+
+for repos_dict in repos_dicts:   #go through all the dictionaries in repos_dicts.
+
+    print('name-:', repos_dict['name'])
+
+    print('owner-:', repos_dict['owner']['login'])
+
+    print('stars-:', repos_dict['stargazers_count'])
+
+    print('repo-:', repos_dict['html_url'])
+
+    print('Description-:', repos_dict['description'])
+
+```
+
+We print the owner of each project, its name, the number of stars it has, the project's description, and its GitHub URL inside the loop:
+
+```bash
+
+name-: public-apis
+
+owner-: public-apis
+
+stars-: 144910
+
+repo-: https://github.com/public-apis/public-apis
+
+Description-: A collective list of free APIs
+
+name-: system-design-primer
+
+owner-: donnemartin
+
+stars-: 139818
+
+repo-: https://github.com/donnemartin/system-design-primer
+
+Description-: Learn how to design large-scale systems.
+
+--snip--
+
+name-: Python
+
+owner-: TheAlgorithms
+
+stars-: 113616
+
+repo-: https://github.com/TheAlgorithms/Python
+
+Description: All Algorithms implemented in Python
+
+```
+
+Let's construct a visual representation of the current popularity of Python works on GitHub, since we have some relevant data. We will create a bar chart: the amount of stars the project has earned will be represented by the height of each bar.
+
+```python
+
+import requests
+
+import pygal
+
+from pygal.style import LightColorizedStyle as LCSe, LightenStyle as LSe
+
+#  Start an API request.
+
+url = 'https://api.github.com/search/repositories?q=language:python&sort=star'
+
+request = requests.get(url)
+
+print("Status  code:", request.status_code)
+
+# In a variable, save the API response.
+
+response_dictionary = request.json()
+
+print("Total-repos:", response_dictionary['total_count'])
+
+# Examine the repository's details.
+
+repos_dicts = response_dictionary['items']
+
+names, stars = [], []
+
+for repos_dict in repos_dicts:
+
+    names.append(repos_dict['name'])
+
+    stars.append(repos_dict['stargazers_count'])
+
+# Make visualization.
+
+mystyle = LSe('#333366', base_style=LCSe)
+
+chart = pygal.Bar(style=mystyle, x_label_rotation=45, show_legend=False)
+
+chart.title = 'Python Projects with the Most Stars on GitHub'
+
+chart.x_labels = names
+
+chart.add('', stars)
+
+chart.render_to_file('py_repos.svg')
+
+```
+
+To begin, we `import pygal` as well as the `Pygal styles` we'll require for the chart. We'll keep printing the status of the API request return and the total amount of repositories detected to see if the API call was successful. We no longer print information about the specific projects that are returned, because that information will be included in the visualization.
```

Comment:
Another small disconnect. Kindly address.
```suggestion
To begin, we `import pygal` as well as the `Pygal styles` we'll require for the chart. We'll keep printing the status of the API request return and the total amount of repositories detected to see if the API call was successful. We no longer print information about the specific projects that are returned, because that information will be included in the visualization.
```